src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,56 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+
+    private static final DateTimeFormatter DATES = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
+    public LocalDate parseDate(String date) {
+        try {
+            return LocalDate.parse(date, DATES);
+        } catch (Exception e) {
+            throw new InvalidDataException(
+                    "Invalid date format. Only dd.MM.yyyy format is accepted"
+            );
+        }
+    }
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        if (names == null || data == null || dateFrom == null || dateTo == null) {
+            throw new InvalidDataException("Invalid data: Please check your input");
+        }
```

Comment:
The null check for input parameters is good, but consider also checking if the `names` and `data` arrays are empty, as this could lead to unexpected behavior or errors.